ExampleMod/Common/GlobalNPCs/ExampleNPCLoot.cs

```diff
@@ -0,0 +1,32 @@
+using Terraria;
+using Terraria.GameContent.ItemDropRules;
+using Terraria.ID;
+using Terraria.ModLoader;
+using ExampleMod.Content.Items;
+
+namespace ExampleMod.Common.GlobalNPCs
+{
+	// This file shows numerous examples of what you can do with the extensive NPC Loot lootable system. 
+	// Despite this file being GlobalNPC, everything here can be used with a ModNPC as well! See examples of this in the Content/NPCs folder.
+	public class ExampleNPCLoot : GlobalNPC
+	{
+		//ModifyNPCLoot uses a unique system called the ItemDropDatabase, which has many different rules for many different drop use cases.
+		//Here we go through all of them, and how they can be used.
+		//There are tons of other examples in vanilla! In a decompiled vanilla build, GameContent/ItemDropRules/ItemDropDatabase adds item drops to every single vanilla NPC, which can be a good resource.
+		public override void ModifyNPCLoot(NPC npc, ItemDropDatabase database) {
+			if ( npc.lifeMax > 5 && npc.value > 0f) { //If npc has health higher than 5 and drops money (aka is not a critter)
+				database.RegisterToNPC(npc.type, ItemDropRule.Common(ModContent.ItemType<ExampleItem>(), 1)); //Make it drop ExampleItem.
+				// ItemDropRule.Common is what you would use in most cases, it simply drops the item with a chance specified.
```

Comment:
I would like to see the reason why we can't have registration look something like the following, be set in stone in this PR's thread.

```cs
npcDrops.Add(ItemDropRule.Common(1, 1));
```

Reply:
Most modded NPC Loot stuff using this method is handled by vanilla with no actual repercussions of doing so (like saving, etc), so making registration different and use a specific method to tmodloader would be difficult, unnecessary, and put modded loot at risk of not doing something vanilla does.

Reply:
Not necessary. I imagined just providing a readonly structure that would have private NPC id and ItemDropDatabase fields, with redirect methods on it.

Reply:
Implemented what I was talking about.